### PR TITLE
javadoc-generation-fix

### PR DESCRIPTION
**Pull request by JockX**
This pull request addresses javadoc linter issues that are curently being worked around by suppressing doclint entirely.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -10,7 +10,6 @@
 
   <properties>
     <java.version>1.6</java.version>
-    <!-- default: empty for compatiblity with Java 8. -->
```
Comment (JockX) [inline pom.xml:13]: This comment obviously refers to a line which is now removed

File: pom.xml [hunk 2]
```diff
@@ -42,8 +41,7 @@
     <plugin.osmaven.version>1.6.2</plugin.osmaven.version>
     <plugin.signature.version>1.1</plugin.signature.version>
     <plugin.surfire.version>3.0.0-M3</plugin.surfire.version>
-    <!-- keep below 3.1.0 because 3.1.0 fails on java 8. -->
-    <plugin.javadoc.version>3.0.1</plugin.javadoc.version>
```
Comment (JockX) [inline pom.xml:46]: The most recent plugin works well with java 8 and above, but reacts more harshly on certain javadoc errors, causing build to fail.

File: pom.xml [hunk 3]
```diff
@@ -217,7 +215,6 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-javadoc-plugin</artifactId>
         <configuration>
-          <doclint>none</doclint>
```
Comment (JockX) [inline pom.xml:220]: After fixing real javadoc problems, the doclint may be restored to its default value, which produces a lot of warnings, but does not fail the build. 
Reasonable alternative setting may perhaps be:
`<doclint>all,-missing</doclint>`

**Comment by tresf** [comment]
From what I understand, the PR does the following:

1. Bumps `plugin.javadoc.version`, fixing JDK8 👍 
2. Fixes `doclint` hard-errors
3. Turns `doclint` back on, consequently adding warnings

Not tested, but no issues.  Thanks!

**Pull request merged**